docs/adr/006_cups.md

```diff
@@ -0,0 +1,296 @@
+
+For increased security the server should "require" client authentication and verify the provided credentials.
+
+Therefore, in order for Basic Station to connect to the CUPS HTTPS endpoint the following files should be provided to the station:
+
+- cups.uri
+- cups.crt (including the certificate in PEM format)
+- cups.key (including the EC Private Key for the child certificate)
+- cups.trust (including the chain of trust certificate for the server-side certificate)
+
+You may find an example on how to generate all the needed certificates in [this script](https://github.com/lorabasics/basicstation/blob/master/examples/cups/prep.sh) from the Basic Station official GitHub repository.
+
+### Authentication to LNS endpoint
+
+As for CUPS, the LNS endpoint should support mutual TLS authentication between client (basic station) and server (network server IoT Edge module).
+
+For increased security the server should "require" client authentication and verify the provided credentials.
+
+Therefore, in order for Basic Station to connect to the LNS WSS endpoint the following files should be provided to the station:
+
+- tc.uri (i.e.: "wss://LoRaWanNetworkSrvModule:5001")
+- tc.crt (including the certificate in PEM format)
+- tc.key (including the EC Private Key for the child certificate)
+- tc.trust (including the chain of trust certificate for the server-side certificate)
+
+Same certificate files as above are expected to be used for the Starter Kit, even though you might want to consider increasing security of the solution by generating different certificates for this endpoint.
+
+There is no need to copy/paste manually these files in the Basic Station; the CUPS Protocol Implementation later described in this document will retrieve the needed credentials from a centralized repository and update the Basic Station configuration when it starts. This option is also providing an option to rotate certificates when these are near the expiration.
+
+### Where to store certificates
+
+There are multiple options of where to store the `cups.{trust,cert,key}` and `tc.{trust,cert,key}` bundle files.
+
+#### a. IoT Hub linked Azure Storage Account
+
+This is the easiest option for a new user of the Starter Kit, even though it's not the safest one.
+
+When deploying the Starter Kit, an Azure Storage Account is linked to the IoT Hub being created for the devices.
+
+LoRa Device Provisioning tool is capable of uploading the CLI provided bundle files and properly update the twin for the concentrator device being created in IoT Hub.
+
+When using this option, the 'Facade' Azure Function will retrieve certificate files from such storage account by using a Managed Identity.
+
+#### b. Azure Key Vault secret
+
+This option requires more manual intervention, as the LoRa Device Provisioning tool is not adapted for uploading the credential bundle files to an Azure Key Vault instance.
```

Comment:
Could we use the function to have additional automation there

Reply:
Can you expand more on the option of having the function as "provisioning proxy endpoint"?

Anyways I would tackle the revisitation of security around function endpoint in a different PR

Reply:
As we discussed, we should not make the function more complex than what it should be. Its purpose should be only the one of downloading the client certificates from Storage or Key Vault.

I will resolve this conversation, feel free to reopen if needed. @Mandur